pkg/resources/database/resource_opensearch.go

```diff
@@ -0,0 +1,618 @@
+package database
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/exoscale/egoscale/v2/oapi"
+
+	"github.com/exoscale/terraform-provider-exoscale/pkg/validators"
+)
+
+type ResourceOpensearchModel struct {
+	// Unknown is a shortcut to avoid implementing Unknownable interface.
+	// TODO: implement Unknownable
+	Unknown bool `tfsdk:"-"`
+
+	ForkFromService          types.String `tfsdk:"fork_from_service"`
+	RecoveryBackupName       types.String `tfsdk:"recovery_backup_name"`
+	IpFilter                 types.Set    `tfsdk:"ip_filter"`
+	KeepIndexRefreshInterval types.Bool   `tfsdk:"keep_index_refresh_interval"`
+	MaxIndexCount            types.Int64  `tfsdk:"max_index_count"`
+	Settings                 types.String `tfsdk:"settings"`
+	Version                  types.String `tfsdk:"version"`
+
+	IndexPatterns []ResourceOpensearchIndexPatternsModel `tfsdk:"index_pattern"`
+	IndexTemplate *ResourceOpensearchIndexTemplateModel  `tfsdk:"index_template"`
+	Dashboards    *ResourceOpensearchDashboardsModel     `tfsdk:"dashboards"`
+}
+
+type ResourceOpensearchIndexPatternsModel struct {
+	MaxIndexCount    types.Int64  `tfsdk:"max_index_count"`
+	Pattern          types.String `tfsdk:"pattern"`
+	SortingAlgorithm types.String `tfsdk:"sorting_algorithm"`
+}
+
+type ResourceOpensearchIndexTemplateModel struct {
+	MappingNestedObjectsLimit types.Int64 `tfsdk:"mapping_nested_objects_limit"`
+	NumberOfReplicas          types.Int64 `tfsdk:"number_of_replicas"`
+	NumberOfShards            types.Int64 `tfsdk:"number_of_shards"`
+}
+
+type ResourceOpensearchDashboardsModel struct {
+	Enabled         types.Bool  `tfsdk:"enabled"`
+	MaxOldSpaceSize types.Int64 `tfsdk:"max_old_space_size"`
+	RequestTimeout  types.Int64 `tfsdk:"request_timeout"`
+}
+
+var ResourceOpensearchSchema = schema.SingleNestedBlock{
+	MarkdownDescription: "*opensearch* database service type specific arguments. Structure is documented below.",
+	Attributes: map[string]schema.Attribute{
+		"fork_from_service": schema.StringAttribute{
+			MarkdownDescription: "❗ Service name",
```

Comment:
```suggestion
			MarkdownDescription: "Service name",
```

Reply:
That is deliberate. Documentation generator adds the symbol for `ForceReplace` in SDK, but for framework and PlanModifier (replacement for `ForceReplace`) it fails to detect it. I added the symbol to keep our documentation consistent.

Reply:
Ah true, I though it was `mandatory` :)